### ISSUE TITLE
Column order update 

### ISSUE THREAD

**Issue by lincolnpahwa** [onechiporenko/ember-models-table#311]
* **I'm submitting a ...**

  - [x] feature request

* **What is the current behavior?**
I couldn't find a way to change _column order_ in the table. i.e. the sequence in which columns are rendered on the page rather than column's data sorting.

* **[Feature] What is the motivation / use case for changing the behavior?**
I want to provide the end user the functionality to select/unselect and order columns of a table and persist user's column preferences. 

* **What exactly you try to do?**
I render the column drop down component as:
`drop-down-template.hbs`
```
{{#drag-sort-list
	items=selectableColumns
	dragEndAction=(action 'columnSortDragEnd')
	as |column|
	}}
{{#content.menu-item }}
  {{#paper-checkbox 
  	value=(not column.isHidden) 
  	onChange=(action '_toggleColumnSelection' column)
  	}}
    <span>{{column.title}}</span>
  {{/paper-checkbox}}
  {{paper-icon "reorder"}}
{{/content.menu-item}}
{{/drag-sort-list}}
```
![image](https://user-images.githubusercontent.com/1823545/39211483-b6177c3c-47d9-11e8-9abc-20885493a789.png)

I am able to achieve the column selection by persisting the `isHidden` values for the given `propertyName` for the columns. I tried reordering the `processedColumns` list and updating the attribute using, in a component that extends `ember-models-table/components/models-table`

`component.js`
```
selectableColumns: filterBy('processedColumns', 'propertyName'),

actions: {
  columnSortDragEnd({ targetList }) {
    set(this, 'processedColumns', targetList);
  }
}
```
but this doesn't update the column order. 
Any help or pointer to get this working is much appreciated.

* **Please tell us about your environment:**

- `ember-models-table` version: 2.5.0
- `ember` version: 3.0.0
- `ember-cli` version:  3.1.2
- `node` version:8.10.0
- `npm` version:5.8.0

* **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. stackoverflow, gitter, etc)
I am using:
*[Ember Drag-Sort](https://github.com/Deveo/ember-drag-sort) for reordering the columns in the column selection drop down menu. 
*[Ember Paper](https://miguelcobain.github.io/ember-paper/) for material design

**Comment by cah-briangantzler** [comment]
The column order is the order the columns are defined in the column array you pass into models-table. If your code would move the columns within this array, the component should re-render with the columns in the new order. (You may have to produce a new array containing the information copied instead of just moving the two indexes, as there would be nothing for ember to invalidate on to detect the change).

When you say persist the order, if you meant the new order would be saved across uses of the application, or if they left the page and came back, you would then put changing the code anyway that built the column array in the new order.

Basically turn your columns property into a computed property that invalidates when the order is changed, and return a new array each time from this computed property.

I think that should do what you want. Let me know

**Comment by lincolnpahwa** [comment]
Yup, I solved it using the indices.

```
columnSortDragEnd({ draggedItem, sourceIndex, targetIndex }) {
	let processedColumns = this.get('processedColumns');

	processedColumns = processedColumns.removeAt(sourceIndex);
	processedColumns.insertAt(targetIndex, draggedItem);
}
```

**Comment by cah-briangantzler** [comment]
The reason you have to move processedColumns by index is because processedColumns is a copy of the original columns array with many more properties added and resolved. Assigning a new array the elements that did not contain this extra internal stuff would not work.

Glad you got your ordering to work, but it seems you are changing it late in the game. Altering the column array passed in I believe would be the more correct way. If the only reason you extended models table was to get the action (access to processedColumns) you would not need to extend models table at all if you created a new columns array.

You may have extended it to get the code to pre-hide the columns. The Columns array has an isHidden attribute on it, you would set this while building the new array in the correct order. Pre hiding and setting the order of the columns is not a concern of models table, it is a concern of the application using models-table.

**Comment by lincolnpahwa** [comment]
I actually extended it take over the column rendering logic and put it in a single place. I have 50+ different grids rendered via models-table with different column definitions and I would have had to go in and add logic in all of them, ( even adding a custom `columns-dropdown` component to the `themeInstance` would have required updating number hbs files). So in order keep the code changes to the minimum against the crunch of time I decided to extend the models-table component. 
Definitely, a custom `columns-dropdown` component is the best to do this.
